src/components/HTMLEngineProvider/BaseHTMLEngineProvider.js

```diff
@@ -70,6 +72,20 @@ function AnchorRenderer({tnode, key, style}) {
     // An auth token is needed to download Expensify chat attachments
     const isAttachment = Boolean(htmlAttribs['data-expensify-source']);
     const fileName = lodashGet(tnode, 'domNode.children[0].data', '');
+    const internalExpensifyPath = lodashGet(htmlAttribs.href.split(CONST.NEW_EXPENSIFY_URL), 1, '');
```

Comment:
NAB: I'm commenting this a NAB because it is very unlikely case, but still I'm commenting about it :)

Don't we want to protect ourselves for the unlikely case of a url containing `CONST.NEW_EXPESIFY_URL` not at the start only? Unlikely cases like: `https://new.expensify.com/?redirect=https://new.expensify.com/something` or `https://someotherdomain.com/?redirect=https://new.expensify.com/settings`

This can be a bit safer done like:
```
if (htmlAttribs.href.startsWith(CONST.NEW_EXPENSIFY_URL)) {
   const internalExpensifyPath = htmlAttribs.href.replace(CONST.NEW_EXPENSIFY_URL, ''); // Remove only first
   ...
}
```

I think it is also more readable than `lodashGet(htmlAttribs.href.split(CONST.NEW_EXPENSIFY_URL), 1, '');`

Reply:
I see your point, but maybe we can wait for this to become a problem (I don't know of any such links that exist yet).

Reply:
>I think it is also more readable

If you have something better to suggest I can improve it! The one you gave wouldn't really get us the "internal path".

```js
'https://blahblah.com'.replace('https://new.expensify.com', ''); // -> 'https://blahblah.com'
```

and 

```js
'https://new.expensify.com/?redirect=https://new.expensify.com/something'.replace('https://new.expensify.com', ''); // -> /?redirect=/something
```

Reply:
I moved the `const internalExpensifyPath = htmlAttribs.href.replace(CONST...` inside the `if` statement, so you would never be calling this on `'https://blahblah.com'`

On the second case, it yields: `"/?redirect=https://new.expensify.com/something"`, not `/?redirect=/something`. The replace function only replaces the first match.

Reply:
Ah I see. Hmm I don't agree it's more readable then since I completely missed what you were going for there :trollface:

Reply:
hahahah, that's a good point. Well, my suggestion is just another developer's opinion, yours may be more readable to you and others. That being settled, I will approve and merge, it works for me.